edifice/hooks.py

```diff
@@ -786,16 +790,64 @@ def expensive_computation():
         bignumber = use_memo(expensive_computation, (x_factor,))
 
     Args:
-        fn: A function of no arguments which returns a value.
-        dependencies: The value will be recomputed when the dependencies change. If :code:`None`, then the value will be recomputed every render.
+        fn:
+            A function of no arguments which returns a value.
+        dependencies:
+            The value will be recomputed when the dependencies change.
+            If :code:`None`, then the value will be recomputed every render.
     Returns:
         The memoized value from calling :code:`fn`.
 
     """
-    stored, _ = use_state([None,None])
+    stored, _ = use_state([None, None])
     if dependencies is None:
-        stored[0] = fn() # type: ignore  # noqa: PGH003
+        stored[0] = fn()  # type: ignore  # noqa: PGH003
     elif stored[1] != dependencies:
-        stored[0] = fn() # type: ignore  # noqa: PGH003
+        stored[0] = fn()  # type: ignore  # noqa: PGH003
         stored[1] = dependencies
-    return stored[0] # type: ignore  # noqa: PGH003
+    return stored[0]  # type: ignore  # noqa: PGH003
+
+
+_T_context = tp.TypeVar("_T_context")
+
+_edifice_context = cast(dict[str, tuple[Any, dict[str, Callable]]], {})
+
+
+def use_context(
+    key: str,
+    value: _T_context | Callable[[], _T_context],
+) -> tuple[_T_context, Callable[[_T_context | Callable[[_T_context], _T_context]], None]]:
+
+    inited, inited_set = use_state([None])
```

Comment:
We have a strong preference for minimal dependencies in Edifice and if we can avoid a `ulid` dependency then I would like to avoid it.

In this case I recommend using the reference of `inited_set` as the unique key. The setter function of a `use_state` Hook is guaranteed to be stable across renders since v2.5.0. https://pyedifice.github.io/versions.html#v2-5-0 ( I see that fact is not documented anywhere else and it should be.)